src/runtime/http.rs

```diff
@@ -15,26 +27,29 @@ impl HttpClient {
     pub(crate) async fn get_and_deserialize_json<'a, T>(
         &self,
         uri: &str,
-        headers: impl IntoIterator<Item = &'a (&'a str, &'a str)>,
-    ) -> reqwest::Result<T>
+        headers: &'a [(&'a str, &'a str)],
+    ) -> Result<T, Box<dyn Error>>
     where
         T: for<'de> Deserialize<'de>,
     {
-        let value = self
-            .request(Method::GET, uri, headers)
-            .await?
-            .json()
+        let res = self
+            .request(Method::GET, Uri::from_str(uri)?, headers)
             .await?;
 
-        Ok(value)
+        let mut buf = body::aggregate(res.into_body()).await?;
+        let mut bytes = vec![0; buf.remaining()];
+        buf.copy_to_slice(&mut bytes);
```

Comment:
Since we're just copying to a slice anyways, we can use `body::to_bytes` directly. Ditto for the other use of `body::aggregate` in `request_and_read_string`.